frontend/packages/kubevirt-plugin/integration-tests/tests/vm.dashboard.scenario.ts

```diff
@@ -0,0 +1,94 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import {
+  addLeakableResource,
+  createResources,
+  removeLeakedResources,
+} from '@console/shared/src/test-utils/utils';
+import { VirtualMachineModel } from '../../src/models';
+import {
+  vmDetailsName,
+  vmDetailsNamespace,
+  vmDetailsNode,
+  vmDetailsIPAddress,
+  vmStatus,
+  vmInventoryNICs,
+  vmInventoryDisks,
+} from '../views/dashboard.view';
+import { getVMManifest, hddDisk, multusNetworkInterface, multusNAD } from './utils/mocks';
+import { VirtualMachine } from './models/virtualMachine';
+import {
+  VM_STATUS,
+  VM_BOOTUP_TIMEOUT_SECS,
+  VM_ACTION,
+  VM_IMPORT_TIMEOUT_SECS,
+  TAB,
+  VM_STOP_TIMEOUT_SECS,
+  NOT_AVAILABLE,
+} from './utils/consts';
+
+describe('Test VM dashboard', () => {
+  const leakedResources = new Set<string>();
+  const testVM = getVMManifest('URL', testName, null, 'foo');
+
+  let vm: VirtualMachine;
+
+  afterAll(async () => {
+    removeLeakedResources(leakedResources);
+  });
+
+  beforeAll(async () => {
+    createResources([multusNAD, testVM]);
+    addLeakableResource(leakedResources, testVM);
+    vm = new VirtualMachine(testVM.metadata);
+    await vm.navigateToDashboard();
+    await browser.wait(
```

Comment:
is there a way to make this condition optional ? meaning if this fails, than continue and do not fail the suite ? In some environments, the import can be so fast that we wont see the transition between Off -> Importing statuses

Reply:
You can use `until.or` and specify multiple conditions. Take a look at this sample. It first waits for `Importing || Off` and then for `Off`, meaning if the VM is being imported first, it will wait at the first wait with adequate timeout. When that finishes, the second wait will block only for a brief moment since the VM should be in Off already.

```
await browser.wait(
  until.or(
    until.textToBePresentInElement(
      vmView.vmDetailStatus(this.namespace, this.name),
      VM_STATUS.Importing,
    ),
    until.textToBePresentInElement(
      vmView.vmDetailStatus(this.namespace, this.name),
      VM_STATUS.Off,
    ),
  ),
  VM_IMPORT_TIMEOUT,
);
await this.waitForStatus(
  VM_STATUS.Off,
  PAGE_LOAD_TIMEOUT_SECS,
);
```

Reply:
after discussion with @rhrazdil we opted for using try/catch